Refactor Controller class (minor changes)

michaelKaefer opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Minor code style changes to Controller class
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | -

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9067)
<!-- Reviewable:end -->

michaelKaefer committed:
Refactor Controller class (minor changes)

Minor code style changes

PierreRambaud commented on classes/controller/Controller.php:
```diff
@@ -192,41 +193,43 @@ public static function getController($class_name, $auth = false, $ssl = false)
+        
         if (
-            !headers_sent() &&
+            ! headers_sent() &&
```
Hum, why this space? There's no rules about it.

PierreRambaud commented on classes/controller/Controller.php:
```diff
@@ -575,23 +578,24 @@ public function getLayout()
      * @throws SmartyException
      */
-    protected function smartyOutputContent($content)
+    protected function smartyOutputContent($templates)
```
Missing doc block modification

michaelKaefer committed:
Removed space, edited doc block

michaelKaefer commented:
@PierreRambaud Thanks! I made another commit.

mickaelandrieu commented on classes/controller/Controller.php:
```diff
@@ -192,32 +193,14 @@ public static function getController($class_name, $auth = false, $ssl = false)
 
     public function __construct()
     {
-        if (is_null($this->display_header)) {
```
I'm not sure of the change you introduce here, because I can't even imagine what the developer wanted to do when he introduced the checks 🤔 

But let's say this was used to set a value for these parameters if they were not defined in children classes:

```php
class ChildrenController extends Controller
{
    public function __construct()
    {
        $this->display_header = false;

        return parent::__construct(); // really common call ^^
    }
}
````

Before your update: ``$this->display_header`` equals ``false``

After your update: ``$this->display_header`` equals ``true``

This is why we shouldn't refactor this specific part, even if this has no impact on Core, but I'll ask review from @Quetzacoalt91 as I'm not really aware of what does this code in the end

michaelKaefer replied:
@mickaelandrieu I did not think about this case. To keep it simple I just reintroduced the is-null-checks. Thank you for the review!

mickaelandrieu reviewed (commented):
needs one more review

michaelKaefer committed:
Reintroduced check if null

michaelKaefer commented:
@mickaelandrieu I already changed everything you asked for (some days ago). I think you can merge it without any more reviews (cause I removed all parts which would have nedded a review)

mickaelandrieu commented:
@michaelKaefer true, I'm kind of really busy today ^^

Thanks !
